http/custom_roundtripper.go

```diff
@@ -0,0 +1,55 @@
+package http
+
+import (
+	"bytes"
+	"context"
+	"io"
+	"log"
+	"net/http"
+	"time"
+
+	awsAuth "github.com/ONSdigital/dp-elasticsearch/v2/awsauth"
+)
+
+type awsSignerRoundTripper struct {
+	signer       *awsAuth.Signer
+	roundTripper http.RoundTripper
+}
+
+func NewAWSSignerRoundTripper(awsFilename, awsProfile, awsRegion, awsService string, customTransport http.RoundTripper) http.RoundTripper {
+	var roundTripper http.RoundTripper
+	if awsRegion == "" || awsService == "" {
+		log.Fatal(context.Background(), "aws region and service should be valid options")
+	}
+	awsSigner, err := awsAuth.NewAwsSigner(awsFilename, awsProfile, awsRegion, awsService)
+	if err != nil {
+		log.Fatal(context.Background(), "failed to create aws v4 signer", err)
```

Comment:
And here also, not returning?

Reply:
The log fatal will log the error and return with an exit code of 1.

Reply:
Are you sure our log.Fatal causes an exit code? I'm not sure it behaves the same a the stand log library.

Reply:
https://pkg.go.dev/log#Fatal

Reply:
Fatal is equivalent to Print() followed by a call to os.Exit(1).

Reply:
// Fatal is equivalent to Print() followed by a call to os.Exit(1).
func Fatal(v ...interface{}) {
	std.Output(2, fmt.Sprint(v...))
	os.Exit(1)
}

Reply:
because we don't have a current context

Reply:
But the log.Fatal function doesn't require a context?

Reply:
but this is done now as , i don't see a point here rather than consistency if passing context as you are annotating the context in anyway.